src/main/java/com/hms_networks/americas/sc/extensions/system/time/SCTimeUtils.java

```diff
@@ -27,4 +52,99 @@ public static String getTimeZoneName() throws Exception {
+   */
+  public static SimpleDateFormat getIso8601UtcTimeFormat() throws IllegalStateException {
+    if (iso8601UtcTimeFormat == null) {
+      throw new IllegalStateException("The local time has not yet been injected.");
+    }
+
+    return iso8601UtcTimeFormat;
+  }
+
+  /**
+   * Returns the simple date format representing ISO 8601 format with the time zone offset appended
+   * (or Z if the timezone is GMT/UTC).
+   *
+   * @return simple date format representing ISO 8601 format with the time zone offset appended
+   * @throws IllegalStateException if the local time has not yet been injected
+   */
+  public static SimpleDateFormat getIso8601LocalTimeFormat() throws IllegalStateException {
+    if (iso8601LocalTimeFormat == null) {
+      throw new IllegalStateException("The local time has not yet been injected.");
+    }
+
+    return iso8601LocalTimeFormat;
+  }
+
+  /**
+   * Returns the designator for the local time zone. (i.e. -05:00, +02:00, Z, etc.)
+   *
+   * @return designator for the local time zone
+   * @throws IllegalStateException if the local time has not yet been injected
+   */
+  public static String getLocalTimeZoneDesignator() throws IllegalStateException {
+    if (localTimeZoneDesignator == null) {
+      throw new IllegalStateException("The local time has not yet been injected.");
+    }
+
+    return localTimeZoneDesignator;
+  }
+
+  /**
+   * Injects the local time into the JVM with the local time offset (from UTC) information from
+   * {@link LocalTimeOffsetCalculator}. A modified time zone with the local time offset (from UTC)
+   * is configured as the JVM default time zone.
+   *
+   * @throws Exception if unable to get the local time zone name
+   */
+  public static void injectJvmLocalTime() throws Exception {
+    // Get current time offset
+    final int calculatedTimeOffsetMilliseconds =
+        (int) LocalTimeOffsetCalculator.calculateLocalTimeOffsetMilliseconds();
+
+    injectCustomJvmLocalTime(calculatedTimeOffsetMilliseconds);
+  }
+
+  /**
+   * Injects the local time into the JVM with the specified local time offset (from UTC) in
+   * milliseconds. A modified time zone with the local time offset (from UTC) is configured as the
+   * JVM default time zone.
+   *
+   * @param timeZoneOffsetMillis local time offset (from UTC) in milliseconds
+   * @throws Exception if unable to get the local time zone name
+   */
+  public static void injectCustomJvmLocalTime(int timeZoneOffsetMillis) throws Exception {
+    // Round time in milliseconds to the nearest 15 minutes
+    final int numberMillisecondsInFifteenMinutes = 900000;
+    int roundedTimeOffsetMillis =
+        Math.round((float) timeZoneOffsetMillis / (float) numberMillisecondsInFifteenMinutes)
+            * numberMillisecondsInFifteenMinutes;
+
+    // Reverse time zone offset (multiply by -1)
+    final int rawTimeZoneOffset = (int) roundedTimeOffsetMillis * -1;
```

Comment:
This is just a question but why are you reversing the offset?  Does this work for both - and + offset zones?

Reply:
We usually refer to the offset *from* local time *to* UTC time. This reverses it so the timezone can be configured with the offset *from* UTC time *to* local time. Yes, it has been tested to work for all time zones

Reply:
I think it warrants documenting that better than, as right now it just reads as an arbitrary reversal.

Reply:
Oh, did not see this comment until now. The page did not auto-refresh like usual

Reply:
Fixed